cmd/tuf/createrepo/main.go

```diff
@@ -105,98 +106,32 @@ func main() {
 
 		if rootok {
 			logging.FromContext(ctx).Infof("Found existing secret config with the TUF root")
-			return
 		}
 		existingSecret.Data = data
 		_, err = clientset.CoreV1().Secrets(ns).Update(ctx, existingSecret, metav1.UpdateOptions{})
 		if err != nil {
 			logging.FromContext(ctx).Fatalf("Failed to update secret %s/%s: %v", ns, *secretName, err)
 		}
 		logging.FromContext(ctx).Infof("Updated secret %s/%s", ns, *secretName)
-		return
-	}
-
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      *secretName,
-		},
-		Data: data,
-	}
-	_, err = clientset.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
-	if err != nil {
-		logging.FromContext(ctx).Fatalf("Failed to create secret %s/%s: %v", ns, *secretName, err)
+	} else {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+				Name:      *secretName,
+			},
+			Data: data,
+		}
+		_, err = clientset.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to create secret %s/%s: %v", ns, *secretName, err)
+		}
+		logging.FromContext(ctx).Infof("Created secret %s/%s", ns, *secretName)
 	}
-	logging.FromContext(ctx).Infof("Created secret %s/%s", ns, *secretName)
-
 	// Serve the TUF repository.
-	fs := http.FileServer(http.Dir("./repo"))
+	fs := http.FileServer(http.Dir(strings.TrimRight(dir, "/")))
```

Comment:
serve the `repository/` subdirectory so that the mirror is the top-level directory